privacyidea/static/components/machine/views/machine.details.html

```diff
@@ -60,34 +60,36 @@ <h3 translate>Tokens and Applications for machine {{ machineid }}</h3>
-                        <b>{{ option | capitalize }}:</b>
-                        {{ token.options[option] }}<br>
+                    <span ng-repeat="(optionname, option) in Applications[token.application].options">
+                        <b>{{ optionname | capitalize }}:</b>
+                        {{ token.options[optionname] }}<br>
                     </span>
-                    <span ng-repeat="option
-                    in Applications[token.application].options.optional">
-                        <b>{{ option | capitalize }}:</b>
-                        {{ token.options[option] }}<br>
-                    </span>
-
                 </td>
+                <!-- Edit options -->
                 <td ng-show="token.optionsEdit">
                     <div class="form-group"
-                         ng-repeat="option in Applications[token.application].options.required">
-                        <label for="option_{{ option }}">{{ option | capitalize }}</label>
-                        <input name="option_{{ option }}"
-                               ng-model="token.options[option]"
-                               class="form-control" ng-required="1"
-                               placeholder="required">
-                    </div>
-                    <div class="form-group"
-                         ng-repeat="option in Applications[token.application].options.optional">
-                        <label for="option_{{ option }}">{{ option | capitalize }}</label>
-                        <input name="option_{{ option }}"
-                               ng-model="token.options[option]"
-                               class="form-control" placeholder="optional">
+                         ng-repeat="(optionname, option)
+                         in Applications[token.application].options">
+                        <label for="option_{{ optionnamme }}">{{ optionname | capitalize }}</label>
+                        <input class="form-control"
+                               ng-required="option.required && !option.value"
+                               ng-model="token.options[optionname]"
+                               ng-show="option.type==='str' && !option.value">
+                        <select class="form-control"
+                                ng-show="option.value"
+                                ng-required="option.required && option.value"
+                                ng-model="token.options[optionname]"
+                                ng-options="value for value in option.value"></select>
+                        <input type="checkbox"
+                               ng-model="token.options[optionname]"
+                               ng-show="option.type==='bool'">
+                        </select>
+                        <p class="help help-block">
+                              {{ option.description }}
+                              {{ option.desc }}
```

Comment:
I think, this can be removed. We don't have any `option.desc` in the code.

Reply:
You are right. But I liked it, because in *some* other cases `desc` is used. So this wey we could have both messages displayed, no matter if the developer uses "description" or the other (faulty) "desc" key.

Reply:
Done in 7257c7ac